### PR TITLE
Feature mes 2034 review driving faults

### PR DESCRIPTION
**Pull request by rossfauldsbjss**
MES 2034 - add ability to add driving fault comments.
Note: the comments text areas should only appear if there are > 15 driving faults and 
no serious or dangerous faults.  In all other cases it should display a summary of the faults 
without the option to edit.

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [x] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

fullsizeoutput_10
fullsizeoutput_f

### CODE REVIEW

File: src/modules/tests/test_data/test-data.actions.ts [hunk 1]
```diff
@@ -96,12 +97,19 @@ export class ControlledStopPending implements Action {
   readonly type = CONTROLLED_STOP_PENDING;
 }
 
+export class AddDrivingFaultComment implements Action {
+  constructor(public competencyName: string, public comment: string) {
+    console.log(`competencyName: ${competencyName}, comment: ${comment}`);
```
Comment (MarkPatt89) [inline src/modules/tests/test_data/test-data.actions.ts]: A sneak wee log left here! :)

Comment (rossfauldsbjss) [reply src/modules/tests/test_data/test-data.actions.ts]: removed.

File: src/pages/debrief/debrief.selector.ts [hunk 2]
```diff
@@ -17,10 +17,31 @@ export const getSeriousOrDangerousFaults = (faults: SeriousFaults | DangerousFau
...
       const label = key as keyof typeof competencyLabels;
-      faultsEncountered.push({ name: fullCompetencyLabels[label], count: value });
+      faultsEncountered.push({ propertyName: key, name: fullCompetencyLabels[label], count: value });
     }
   });
   return faultsEncountered.sort((a, b) => b.count - a.count);
 };
+
+export const displayDrivingFaultComments = (data: TestData) : boolean => {
+  const seriousFaults =  getSeriousOrDangerousFaults(data.seriousFaults);
```
Comment (szabolcs-keresztes) [inline src/pages/debrief/debrief.selector.ts]: Why don't we have 2 separate functions, `getSeriousFaults` and `getDangerousFaults`?

File: src/modules/tests/test_data/test-data.actions.ts [hunk 3]
```diff
@@ -22,6 +22,7 @@ export const TOGGLE_PLANNING_ECO = '[Eco] Toggle Planning Eco';
 export const TOGGLE_CONTROLLED_STOP = '[ControlledStop] Toggle Controlled Stop';
 export const CONTROLLED_STOP_COMPLETE = '[ControlledStop] Controlled Stop Complete';
 export const CONTROLLED_STOP_PENDING = '[ControlledStop] Controlled Stop Pending';
+export const ADD_DRIVING_FAULT_COMMENT = '[Competency] Add driving fault comment';
```
Comment (szabolcs-keresztes) [inline src/modules/tests/test_data/test-data.actions.ts]: Does the `Competency` component dispatch this action?

File: src/pages/office/components/drivingFaultComment/driving-fault-comment.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      debounceTime(1000),
+      distinctUntilChanged(),
+    );
+    const subscription = changeStream$
+      .subscribe((newVal: string) => this.store$.dispatch(
+        new actionType(this.extractPropertyNameFromId(id), newVal)));
+    return subscription;
+  }
+
+  extractPropertyNameFromId(id: any) {
```
Comment (szabolcs-keresztes) [inline src/pages/office/components/drivingFaultComment/driving-fault-comment.ts]: This function should be part of a different module. DrivingFaultCommentComponent has no responsibility of extracting property name from id.

File: src/pages/office/office.html [hunk 5]
```diff
@@ -274,71 +276,129 @@ <h4>Test Summary</h4>
...
+              <driving-faults-badge class="counter driving-faults" [count]="drivingFault.count"></driving-faults-badge>
+            </div>
+          </ion-col>
+          <ion-col class="component-label" col-58 align-self-center>
+            <label class="fault-label">{{drivingFault.name}}</label>
+          </ion-col>
+        </ion-row>
+      </ion-card-content>
+    </ion-card>
+  </form>
```
Comment (szabolcs-keresztes) [inline src/pages/office/office.html]: This file is 410 lines. Can you please split it into multiple components?

File: src/pages/office/office.ts [hunk 6]
```diff
@@ -235,8 +249,35 @@ export class OfficePage extends BasePageComponent {
         select(getTestData),
         map(data => getSeriousOrDangerousFaults(data.dangerousFaults)),
       ),
+      drivingFaults$: this.store$.pipe(
```
Comment (szabolcs-keresztes) [inline src/pages/office/office.ts]: Can you please extract the `testData` into it's own observable and reuse it in these different selections?

File: src/pages/office/office.html (hunk 5 above)
Comment (johnmorleybjss) [reply src/pages/office/office.html]: I have been working on breaking this component up into a component per row, this change will come soon. 👍

Comment (MarkPatt89) [reply src/pages/office/office.html]: That would be outside of the scope of this ticket. Ross has made a component for the driving faults. John is currently working on adding components for other areas.

File: src/pages/office/office.ts (hunk 6 above)
Comment (johnmorleybjss) [reply src/pages/office/office.ts]: Done

File: src/pages/office/office.html (hunk 5 above)
Comment (szabolcs-keresztes) [reply src/pages/office/office.html]: Okay, nice one!

File: src/modules/tests/test_data/test-data.actions.ts (hunk 3 above)
Comment (MarkPatt89) [reply src/modules/tests/test_data/test-data.actions.ts]: Done

File: src/pages/debrief/debrief.selector.ts (hunk 2 above)
Comment (MarkPatt89) [reply src/pages/debrief/debrief.selector.ts]: Implemented in mes-2033.

**Pull request merged**